models/contacts.js

```diff
@@ -1,19 +1,115 @@
-// const fs = require('fs/promises')
+import * as path from 'path';
+import { readFile, writeFile } from 'fs/promises';
+import { nanoid } from 'nanoid';
 
-const listContacts = async () => {}
+const contactsPath = path.resolve('./models', 'contacts.json');
 
-const getContactById = async (contactId) => {}
+const readContactsFile = async () => {
+  try {
+    const contactsJson = await readFile(contactsPath);
+    const contacts = JSON.parse(contactsJson);
+    return contacts;
+  } catch (err) {
+    console.error('Error reading contacts from file: ', err);
+    throw err;
+  }
+};
 
-const removeContact = async (contactId) => {}
+const writeContactsFile = async data => {
+  try {
+    await writeFile(contactsPath, JSON.stringify(data, null, 2));
+  } catch (err) {
+    console.error('Error writing contacts to file: ', err);
+    throw err;
+  }
+};
 
-const addContact = async (body) => {}
+export const listContacts = async () => {
+  try {
+    const contacts = await readContactsFile();
+    if (contacts) return contacts;
+    if (!contacts) console.log(`No contacts found`);
+    return false;
+  } catch (err) {
+    console.log('Error getting contact list: ', err);
+    throw err;
+  }
+};
 
-const updateContact = async (contactId, body) => {}
+export const getContactById = async contactId => {
+  try {
+    const contacts = await listContacts();
+    const index = contacts.findIndex(contact => contact.id === contactId);
+    if (index === -1) {
+      console.log(`There is no contact with id ${contactId} !`);
+      return false;
+    }
+    return contacts[index];
+  } catch (err) {
+    console.log(`Error getting contact with id ${contactId}: `, err);
+    throw err;
+  }
+};
 
-module.exports = {
-  listContacts,
-  getContactById,
-  removeContact,
-  addContact,
-  updateContact,
-}
+export const removeContact = async contactId => {
+  try {
+    const contacts = await listContacts();
+    const index = contacts.findIndex(contact => contact.id === contactId);
+    if (index === -1) {
+      console.log(`There is no contact with id ${contactId} !`);
+      return false;
+    }
+    contacts.splice(index, 1);
+    await writeContactsFile(contacts);
+    console.log(`Contact with id ${contactId} removed successfully`);
+    return true;
+  } catch (err) {
+    console.log(`Error removing contact with id ${contactId}: `, err);
+    throw err;
+  }
+};
+
+export const addContact = async body => {
+  try {
+    const contacts = await listContacts();
+    if (contacts) {
```

Comment:
you always should get at least empty array from `listContacts()` method.
`listContacts()` is actually used for enpoint. Would be better if you use `readContactsFile` here which do the same :).
Still personally i would name it `getContacts`

Reply:
I mean you dont need this `if`